core/templates/directives/mathjax.directive.ts

```diff
@@ -23,30 +23,49 @@ import {
   ElementRef,
   Input,
   OnChanges,
+  OnInit,
   SimpleChanges,
 } from '@angular/core';
+import {
+  InsertScriptService,
+  KNOWN_SCRIPTS,
+} from 'services/insert-script.service';
 
 @Directive({
   selector: '[oppiaMathJax]',
 })
-export class MathJaxDirective implements OnChanges {
+export class MathJaxDirective implements OnChanges, OnInit {
   // This property is initialized using Angular lifecycle hooks
   // and we need to do non-null assertion. For more information, see
   // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1 .
   @Input('oppiaMathJax') texExpression!: string;
 
-  constructor(private el: ElementRef) {}
+  loaded = false;
+  constructor(
+    private el: ElementRef,
+    private insertScriptService: InsertScriptService
+  ) {}
 
+  ngOnInit(): void {
+    this.insertScriptService.loadScript(KNOWN_SCRIPTS.MATHJAX, () => {
+      this.loaded = true;
+      this.renderExpression();
+    });
+  }
   ngOnChanges(changes: SimpleChanges): void {
     if (
+      this.loaded &&
```

Comment:
The "loaded" flag here can be a bit confusing. It looks like the idea is to ensure that the script has loaded before executing anything in `ngOnChanges`. If so, could we add a method in InsertScriptService that will check if the script has loaded and return a boolean (e.g. checks that the script is present in `fullyLoadedScripts`) and call that here instead? Ditto in MathExpressionContentEditorComponent below.

Reply:
Done